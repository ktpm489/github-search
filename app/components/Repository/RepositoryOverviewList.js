import React from 'react';
import PropTypes from 'prop-types';
import { FlatList, StyleSheet, View } from 'react-native';

// eslint-disable-next-line import/no-named-as-default
import RepositoryOverview from './RepositoryOverview';

class RepositoryOverviewList extends React.Component {
  static propTypes = {
    repos: PropTypes.arrayOf(
      PropTypes.shape({
        node: PropTypes.shape({
          name: PropTypes.string.isRequired,
          description: PropTypes.string,
          url: PropTypes.string.isRequired,
        }),
      })
    ).isRequired,
    navigation: PropTypes.object.isRequired,
    onEndReached: PropTypes.func.isRequired,
  };

  renderItem = ({
    item: {
      node: { name, description, url },
    },
  }) => {
    const { navigation } = this.props;
    return (
      <RepositoryOverview
        name={name}
        description={description}
        url={url}
        navigation={navigation}
      />
    );
  };

  render() {
    const { repos, onEndReached } = this.props;
    return (
      <FlatList
        style={{ flex: 1 }}
        data={repos}
        renderItem={(repo) => this.renderItem(repo)}
        keyExtractor={(item, index) => index}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
        onEndReached={onEndReached}
      />
    );
  }
}

const styles = StyleSheet.create({
  separator: {
    height: 1,
    backgroundColor: '#222',
    flex: 1,
    flexDirection: 'row',
  },
});

export default RepositoryOverviewList;
