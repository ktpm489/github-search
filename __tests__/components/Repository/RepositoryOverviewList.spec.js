import React from 'react';
import { shallow } from 'enzyme';

import RepositoryOverviewList from '../../../app/components/Repository/RepositoryOverviewList';

describe('<RepositoryOverviewList />', () => {
  const repos = [
    {
      node: {
        name: 'd3-react-map',
        description:
          'A mapping app visualizing police shootings in the United States.',
        url: 'https://github.com/parkerziegler/d3-react-map',
      },
    },
    {
      node: {
        name: 'github-search',
        description: 'A React Native app for searching Github users.',
        url: 'https://github.com/parkerziegler/github-search',
      },
    },
  ];

  it('renders correctly with all required props', () => {
    const onEndReached = jest.fn();
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(
      <RepositoryOverviewList
        repos={repos}
        onEndReached={onEndReached}
        navigation={navigation}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
