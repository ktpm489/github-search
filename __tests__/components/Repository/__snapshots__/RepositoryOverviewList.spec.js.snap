// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryOverviewList /> renders correctly with all required props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryOverviewList
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    onEndReached={[MockFunction]}
    repos={
      Array [
        Object {
          "node": Object {
            "description": "A mapping app visualizing police shootings in the United States.",
            "name": "d3-react-map",
            "url": "https://github.com/parkerziegler/d3-react-map",
          },
        },
        Object {
          "node": Object {
            "description": "A React Native app for searching Github users.",
            "name": "github-search",
            "url": "https://github.com/parkerziegler/github-search",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ItemSeparatorComponent": [Function],
      "data": Array [
        Object {
          "node": Object {
            "description": "A mapping app visualizing police shootings in the United States.",
            "name": "d3-react-map",
            "url": "https://github.com/parkerziegler/d3-react-map",
          },
        },
        Object {
          "node": Object {
            "description": "A React Native app for searching Github users.",
            "name": "github-search",
            "url": "https://github.com/parkerziegler/github-search",
          },
        },
      ],
      "disableVirtualization": false,
      "horizontal": false,
      "initialNumToRender": 10,
      "keyExtractor": [Function],
      "maxToRenderPerBatch": 10,
      "numColumns": 1,
      "onEndReached": [MockFunction],
      "onEndReachedThreshold": 2,
      "renderItem": [Function],
      "scrollEventThrottle": 50,
      "style": Object {
        "flex": 1,
      },
      "updateCellsBatchingPeriod": 50,
      "windowSize": 21,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ItemSeparatorComponent": [Function],
        "data": Array [
          Object {
            "node": Object {
              "description": "A mapping app visualizing police shootings in the United States.",
              "name": "d3-react-map",
              "url": "https://github.com/parkerziegler/d3-react-map",
            },
          },
          Object {
            "node": Object {
              "description": "A React Native app for searching Github users.",
              "name": "github-search",
              "url": "https://github.com/parkerziegler/github-search",
            },
          },
        ],
        "disableVirtualization": false,
        "horizontal": false,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReached": [MockFunction],
        "onEndReachedThreshold": 2,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "style": Object {
          "flex": 1,
        },
        "updateCellsBatchingPeriod": 50,
        "windowSize": 21,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
